addons/mail/static/src/messaging/component/composer_text_input/composer_text_input.js

```diff
@@ -144,10 +178,82 @@ class ComposerTextInput extends Component {
         this._textareaRef.el.style.height = (this._textareaRef.el.scrollHeight) + "px";
     }
 
+    /**
+     * Validates user's current typing as a correct mention keyword
+     * in order to trigger mentions suggestions display.
+     * Returns the mention keyword without the delimiter if it has been validated
+     * and false if not
+     *
+     * @private
+     * @param delimiter
+     * @param beginningOnly
+     * @returns {string|boolean}
+     */
+    _validateMentionKeyword(delimiter, beginningOnly) {
+        const leftString = this.getContent().substring(0, this._getSelectionStart());
+
+        // use position before delimiter because there should be whitespaces
+        // or line feed/carriage return before the delimiter
+        const beforeDelimiterPosition = leftString.lastIndexOf(delimiter) - 1;
+        if (beginningOnly && beforeDelimiterPosition > 0) {
+            return false;
+        }
+        let searchStr = this.getContent().substring(beforeDelimiterPosition, this._getSelectionStart());
+        // regex string start with delimiter or whitespace then delimiter
+        const pattern = "^"+delimiter+"|^\\s"+delimiter;
+        const regexStart = new RegExp(pattern, 'g');
+        // trim any left whitespaces or the left line feed/ carriage return
+        // at the beginning of the string
+        searchStr = searchStr.replace(/^\s\s*|^[\n\r]/g, '');
+        if (regexStart.test(searchStr) && searchStr.length) {
+            searchStr = searchStr.replace(pattern, '');
+            return !searchStr.includes(' ') && !/[\r\n]/.test(searchStr)
+                ? searchStr.replace(delimiter, '')
+                : false;
+        }
+        return false;
+    }
+
+    /**
+     * Detects if mentioned partners are still in the composer text input
+     * and removes them if not.
+     *
+     * @private
+     */
+    _validatePartnerIds() {
+        const inputMentions = this.getContent().match(new RegExp("@"+'[^ ]+(?= |&nbsp;|$)', 'g'));
+        if (inputMentions) {
+            for (const mention of this.composer.mentionedPartners) {
+                let inputMention = inputMentions.find(item => {
+                    return item === ("@" + mention.name).replace(/ /g, '\u00a0');
+                });
+                if (!inputMention) {
+                    this.composer.update({ mentionedPartners: [['link', mention]] });
```

Comment:
It doesn't look like this code what you said.
From my understanding: it (re-)links partners that are already mentioned and which haven't be found in content.
This doesn't make sense, and the slight change that would is to unlink mentioned partner that are no longer mentioned.

If the intent is what you just said, then I don't think we want this, because this would consist of finding whether a mention in content maps to a partner, and this cannot be done in a reliable way (mentions in content are basically partner names, which are not unique).

However, removing mentioned partners that are no longer in content make totally sense IMO.

Reply:
Perfect ! You didn't fall into my trap :p 
I was wrong indeed.